apis/v1alpha2/validation/gateway.go

```diff
@@ -44,7 +45,32 @@ func validateGatewaySpec(spec *gatewayv1a2.GatewaySpec, path *field.Path) field.
 // to the Gateway API specification.
 func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	// TODO [danehans]: Add additional validation of listener fields.
-	return validateListenerHostname(listeners, path)
+	var errs field.ErrorList
+	e1 := validateListenerTLSConfig(listeners, path)
+	errs = append(errs, e1...)
+	e2 := validateListenerHostname(listeners, path)
+	errs = append(errs, e2...)
+	return errs
+}
+
+func validateListenerTLSConfig(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	nonTlsProtocols := []gatewayv1a2.ProtocolType{gatewayv1a2.HTTPProtocolType, gatewayv1a2.UDPProtocolType, gatewayv1a2.TCPProtocolType}
```

Comment:
Not sure if this would be used globally by  other functions too

Reply:
Thanks for your review

Reply:
@tokers can we discuss as why do you think it should be placed outside ? I am fairly new to this project so I am  not aware of specific usecases where this would be helpful

Reply:
The `nonTlsProtocols` is composed of constants, so it can be created before the function call, and we just read it, we don't have to create multiple copies when we enter this function, this is also good for the performance (but maybe it's trivial).

Reply:
I think moving this and the tcpUdpProtocols outside would be for the best because I expect other validations to use these as well.

Also, once you move them outside, replace the `contains()` with:
```go
func isProtocolInSubset(protocol gatewayv1a2.ProtocolType, set []gatewayv1a2.ProtocolType) bool {}
```

Reply:
Thanks @tokers @hbagdi  will fix this

Reply:
Fixed